query/src/main/scala/filodb/query/exec/AggrOverRangeVectors.scala

```diff
@@ -513,18 +515,30 @@ class TopBottomKRowAggregator(k: Int, bottomK: Boolean) extends RowAggregator {
     val colSchema = Seq(ColumnInfo("timestamp", ColumnType.LongColumn), ColumnInfo("value", ColumnType.DoubleColumn))
     val recSchema = SerializableRangeVector.toSchema(colSchema)
     val resRvs = mutable.Map[RangeVectorKey, RecordBuilder]()
-    // We limit the results wherever it is materialized first. So it is done here.
-    aggRangeVector.rows.take(limit).foreach { row =>
-      var i = 1
-      while(row.notNull(i)) {
-        val rvk = CustomRangeVectorKey.fromZcUtf8(row.filoUTF8String(i))
-        val builder = resRvs.getOrElseUpdate(rvk, SerializableRangeVector.toBuilder(recSchema))
-        builder.startNewRecord()
-        builder.addLong(row.getLong(0))
-        builder.addDouble(row.getDouble(i + 1))
-        builder.endRecord()
-        i += 2
+    // Important TODO / TechDebt: We need to replace Iterators with cursors to better control
+    // the chunk iteration, lock acquisition and release. This is much needed for safe memory access.
+    try {
+      // validate no locks are held by the thread. If there are, quite possible a lock acquire or release bug exists
+      val numLocksReleased = OffheapLFSortedIDMap.releaseAllSharedLocks()
```

Comment:
Add a common utility method in OffheapLFSortedIDMap, validateNoSharedLocks, which does the release and warning.